tests/0089-max_poll_interval.c

```diff
@@ -440,7 +452,8 @@ do_test_rejoin_after_interval_expire(rd_bool_t forward_to_another_q) {
 int main_0089_max_poll_interval(int argc, char **argv) {
         do_test();
         do_test_with_log_queue();
-        do_test_rejoin_after_interval_expire(rd_false);
-        do_test_rejoin_after_interval_expire(rd_true);
+        do_test_rejoin_after_interval_expire(rd_false, rd_false);
+        do_test_rejoin_after_interval_expire(rd_true, rd_false);
+        do_test_rejoin_after_interval_expire(rd_false, rd_true);
```

Comment:
I think we should add a test for the specific case that brought this bug to light, too, which uses `rd_kafka_conf_set_consume_cb` to set a consume_cb, and the calls `rd_kafka_poll`.

Reply:
Added a new test, though `test_wait_event` is unable to capture RD_KAFKA_EVENT_REBALANCE not sure why, so haven't added any assert in the test. But added a poll, which should fail if the consumer has left the group.